src/runtime/sitemap/urlset/filter.ts

```diff
@@ -14,8 +17,17 @@ function createFilter(options: CreateFilterOptions = {}): (path: string) => bool
     return () => true
 
   return function (path: string): boolean {
+    const regexPattern = /\/(.*?)\/([gimsuy]*)$/
     for (const v of [{ rules: exclude, result: false }, { rules: include, result: true }]) {
-      const regexRules = v.rules.filter(r => r instanceof RegExp) as RegExp[]
+      const regexRules = (v.rules.filter(r => typeof r === 'object' && r.regex && typeof r.regex === 'string') as RegexObjectType[])
+                                 .map((r) => {
+                                  const match = r.regex.match(regexPattern)
+                                  if (!match || match.length < 3){
+                                    console.warn(`Invalid regex rule: ${r.regex}`)
```

Comment:
i think its preferrable to just throw an exception if the regex is invalid, if we do want to provide extra DX around this, we should do it at the normalisation step with module.ts